config/config.go

```diff
@@ -47,10 +47,14 @@ type SourceSubnet struct {
 	WSEndpoint            string                           `mapstructure:"ws-endpoint" json:"ws-endpoint"`
 	MessageContracts      map[string]MessageProtocolConfig `mapstructure:"message-contracts" json:"message-contracts"`
 	SupportedDestinations []string                         `mapstructure:"supported-destinations" json:"supported-destinations"`
-	StartBlockHeight      uint64                           `mapstructure:"start-block-height" json:"start-block-height"`
+	StartBlockHeight      int64                            `mapstructure:"start-block-height" json:"start-block-height"`
 
 	// convenience field to access the supported destinations after initialization
 	supportedDestinations set.Set[ids.ID]
+
+	// convenience field to store the starting block height as a bigint so that we can differentiate
+	// between a 0 value and an unset value
+	StartBlockHeightBigInt *big.Int
```

Comment:
We should probably make `StartBlockHeight` private so that it's not mistakenly used somewhere.

Reply:
I believe it needs to be public so that we can unmarshal from JSON into the struct

Reply:
Oh hmm seems like you're right.